app/Http/Controllers/PhotoAdminController.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Proto\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use Proto\Http\Requests;
+use Proto\Http\Controllers\Controller;
+use Proto\Models\Photo;
+use Proto\Models\PhotoManager;
+use Proto\Models\PhotoAlbum;
+use Proto\Models\PhotoLikes;
+use Auth;
+
+use Proto\Models\StorageEntry;
+use Redirect;
+
+class PhotoAdminController extends Controller
+{
+    public function index()
+    {
+        return view('photos.admin.index', ['query' => '']);
+    }
+
+    public function search(Request $request)
+    {
+        return view('photos.admin.index', ['query' => $request->input('query')]);
+    }
+
+    public function create(Request $request)
+    {
+        $album = new PhotoAlbum();
+        $album->name = $request->input('name');
+        $album->date_taken = strtotime($request->input('date'));
+        if ($request->input('private')) {
+            $album->private = true;
+        }
+        $album->save();
+
+        return redirect(route('photo::admin::edit', ['id' => $album->id]));
+    }
+
+    public function edit($id)
+    {
+
+        $photos = PhotoManager::getPhotos($id);
+
+        if ($photos) return view('photos.admin.edit', ['photos' => $photos]);
+
+        abort(404);
+    }
+
+    public function update(Request $request, $id)
+    {
+        $album = PhotoAlbum::find($id);
+        $album->name = $request->input('album');
+        $album->date_taken = strtotime($request->input('date'));
+        if ($request->input('private')) {
+            $album->private = true;
+        } else {
+            $album->private = false;
+        }
+        $album->save();
+        return redirect(route('photo::admin::edit', ['id' => $id]));
+    }
+
+    public function upload(Request $request, $id)
+    {
+        $album = PhotoAlbum::find($id);
+        $response = "ERROR";
+        if ($request->has('file') && !$album->published) {
+            $uploadFile = $request->file('file');
+
+            $photo = $this->createPhotoFromUpload($uploadFile, $id);
+
+            $response = view('website.layouts.macros.selectablephoto', ['photo' => $photo]);
+        }
+        return $response;
+    }
+
+    public function action(Request $request, $id)
```

Comment:
Why one function and not three different functions?

Reply:
We both can't think of a better way to do this.